ckanext/stcndm/validators.py

```diff
@@ -28,6 +28,25 @@ def scheming_validator(fn):
     fn.is_a_scheming_validator = True
     return fn
 
+def repeating_text_delimited(key, data, errors, context):
+    if errors[key]:
+        return
+
+    value = data[key]
+    if isinstance(value, list):
+        return
+    else:
+        values = value.split(';')
+        out = []
+
+        for val in values:
+            val = val.strip()
+
+            if (val):
+                out.append(val)
+
+        data[key] = json.dumps(out)
```

Comment:
So if the value is a list it stays a list, but if it's a string it gets split and converted to a JSON string containing a list? Why not store the list if a list is valid (it would be idempotent that way too)